src/v/kafka/server/group.cc

```diff
@@ -1063,7 +1066,8 @@ void group::complete_join() {
             ssx::background
               = store_group(checkpoint(assignments_type{}))
                   .then(
-                    []([[maybe_unused]] result<raft::replicate_result> r) {});
+                    [this_group = shared_from_this()](
```

Comment:
The observed crash was at the other location (above), but I found this background task too.

The issue is less clear here: we do not capture `this`, but internally `store_group` calls into a method on the `_partition` member which calls into the `_raft` member: the lifetime of both of these objects potentially depends on the group object: it does seem like they can be destroyed if the group goes away and other conditions are met. Keeping the group alive seems like a safe choice here: this will also keep the paritoin and consensus_ptr alive.